src_compressible/flux.cpp

```diff
@@ -440,6 +677,17 @@ void calculateFlux(const MultiFab& cons_in, const MultiFab& prim_in,
                 Real phiflxshear = wiener[1]*(prim(i,j-1,k,1)+prim(i,j,k,1)) +
                                    wiener[3]*(prim(i,j-1,k,3)+prim(i,j,k,3));
 
+                if ((j == 0) and (bc_mass_lo[1] != 3) and (bc_mass_lo[1] != -1)) { // ignore for reservoirs and periodic BC
+                    phiflxdiag =  2.0*wiener[1]*prim(i,j-1,k,1);
+                    phiflxshear = 2.0*wiener[2]*prim(i,j-1,k,2) +
+                                  2.0*wiener[3]*prim(i,j-1,k,3);
+                }
+                if ((j == n_cells[1]) and (bc_mass_hi[1] != 3) and (bc_mass_hi[1] != -1)) { // ignore for reservoirs and periodic BC
+                    phiflxdiag =  2.0*wiener[1]*prim(i,j,k,1);
+                    phiflxshear = 2.0*wiener[2]*prim(i,j,k,2) +
+                                  2.0*wiener[3]*prim(i,j,k,3);
+                }
```

Comment:
These and the z-case aren't correct.  Here phiflxdiag should be associated with wiener[2]

Reply:
Correct -- fixed it.